### DISCUSSION TITLE
convert int? to int in flutter

### DISCUSSION THREAD

**Discussion by harsh4248** [community/community#20151] [Codespaces] [answered]
i get int? variable from SharedPreferences class with getInt() method. I have to put this int? as a parameter for Duration Object but i need to have an int. What can I do? This is the code:

```
int? duration = ParametersHelper.getDuration();
Future.delayed(const Duration(milliseconds: duration));
```

This is getDuration() method in ParametersHelper:

```
static int? getDuration() {
    return _pref.getInt("Duration");
  }
```

**Comment by chiragmehta900** [comment, marked as answer]
You can either override this behavior by explicitly telling the dart compiler that although the variable can be nullable I guarantee that it's not null, be careful when using it though.

```
int duration = ParametersHelper.getDuration()!; // using the ! operator
Future.delayed(const Duration(milliseconds: duration));
```

The second option is to use the ?? operator which means if the value is null assign another value

```
int duration = ParametersHelper.getDuration() ?? 2; // if null assign 2
Future.delayed(const Duration(milliseconds: duration));
```

**Comment by hashfx** [comment]
Hey @harsh4248, are you using Flutter in codespaces? If Yes, would you share the steps for using Flutter on codespace!
